src/Nethermind/Nethermind.Specs/Forks/15_Shanghai.cs

```diff
@@ -16,73 +16,22 @@
+        protected Shanghai()
         {
         }
 
-        public static IReleaseSpec Instance => LazyInitializer.EnsureInitialized(ref _instance, () => new Shanghai());
+        public new static IReleaseSpec Instance => LazyInitializer.EnsureInitialized(ref _instance, () => new Shanghai());
 
-        public string Name => "Shanghai";
-        public long MaximumExtraDataSize => 32;
-        public long MaxCodeSize => 24576;
-        public long MinGasLimit => 5000;
-        public long GasLimitBoundDivisor => 0x0400;
-        public UInt256 BlockReward { get; } = UInt256.Parse("2000000000000000000");
-        public long DifficultyBombDelay => 10700000L;
-        public long DifficultyBoundDivisor => 0x0800;
-        public long? FixedDifficulty => null;
-        public int MaximumUncleCount => 2;
-        public bool IsTimeAdjustmentPostOlympic => true;
-        public bool IsEip2Enabled => true;
-        public bool IsEip7Enabled => true;
-        public bool IsEip100Enabled => true;
-        public bool IsEip140Enabled => true;
-        public bool IsEip150Enabled => true;
-        public bool IsEip155Enabled => true;
-        public bool IsEip158Enabled => true;
-        public bool IsEip160Enabled => true;
-        public bool IsEip170Enabled => true;
-        public bool IsEip196Enabled => true;
-        public bool IsEip197Enabled => true;
-        public bool IsEip198Enabled => true;
-        public bool IsEip211Enabled => true;
-        public bool IsEip214Enabled => true;
-        public bool IsEip649Enabled => true;
-        public bool IsEip658Enabled => true;
-        public bool IsEip145Enabled => true;
-        public bool IsEip1014Enabled => true;
-        public bool IsEip1052Enabled => true;
-        public bool IsEip1283Enabled => false;
-        public bool IsEip1234Enabled => true;
-        public bool IsEip1344Enabled => true;
-        public bool IsEip2028Enabled => true;
-        public bool IsEip152Enabled => true;
-        public bool IsEip1108Enabled => true;
-        public bool IsEip1884Enabled => true;
-        public bool IsEip2200Enabled => true;
-        public bool IsEip2315Enabled => false;
-        public bool IsEip2537Enabled => false;
-        public bool IsEip2565Enabled => true;
-        public bool IsEip2929Enabled => true;
-        public bool IsEip2930Enabled => true;
-        public bool IsEip158IgnoredAccount(Address address) => false;
-        public bool IsEip1559Enabled => true;
-        public bool IsEip3198Enabled => true;
-        public bool IsEip3529Enabled => true;
-        public bool IsEip3541Enabled => true;
-        public bool IsEip3607Enabled => true;
-        public bool IsEip3675Enabled => false;
-        public long Eip1559TransitionBlock => 12965000;
-        public bool IsEip1153Enabled => true;
+        public override string Name => "Shanghai";
+        public override bool IsEip1153Enabled => true;
+        public override bool IsEip3675Enabled => true;
```

Comment:
Another way is to set in the constructor, to avoid virtual calls. Do we call this props a lot?

Reply:
according to @LukaszRozmej they will be virtual calls because they are called through the interface anyways. According to @tkstanczak yes we do call these props a lot in the EVM. Benchmark was created. no real perf hit was noticed.